models/agent_helpers.go

```diff
@@ -209,6 +211,34 @@ func FindPMMAgentsForService(q *reform.Querier, serviceID string) ([]*Agent, err
 	return res, nil
 }
 
+// FindPMMAgentsForVersion selects pmm-agents with version >= minPMMAgentVersion.
+func FindPMMAgentsForVersion(logger *logrus.Entry, agents []*Agent, minPMMAgentVersion *version.Parsed) []*Agent {
+	if len(agents) == 0 {
+		return nil
+	}
+
+	if minPMMAgentVersion == nil {
+		return agents
```

Comment:
In line 223 you are creating (and then returning) a copy of the in `agents` arrays.
Does returning the same arrays affects the rest of the code anyway? (like in an append?)

Reply:
No, it shouldn't